bayes_opt/util.py

```diff
@@ -99,6 +99,9 @@ def utility(self, x, gp, y_max):
             return self._ei(x, gp, y_max, self.xi)
         if self.kind == 'poi':
             return self._poi(x, gp, y_max, self.xi)
+        
+    def decay_kappa(self, decay_factor):
```

Comment:
I meant passing the `decay_factor` (and `_delay` too) to the constructors, and then calling `util.update_params()` (or similar name) in the loop. All this logic `if kappa_decay != 1 and max(0, len(self.res) - init_points) - kappa_decay_delay > 0:` could reside inside the `UtilityFunction` object.

Reply:
Ah i see. I will make the changes. For delay to be used however we will need to feed in the # of BO iters to the update function. Lmk how it looks now.

Reply:
`UtilityFunction` could keep an internal counter to keep track of delay.

Reply:
ok done